CoreUtilities/src/au/gov/asd/tac/constellation/utilities/genericjsonio/JsonIO.java

```diff
@@ -0,0 +1,289 @@
+     * directory. If filePrefix was provided as a non empty string, then only
+     * files prefixed with this value are available to the user to load.
+     *
+     * @param loadDir the directory location of the JSON files as a string
+     * @param filePrefix if not blank a prefix string is pre-pended to the
+     * beginning of the filename this can be used to create categorized JSON
+     * files with forced name groupings
+     *
+     * @return The JsonNode of the selected preference or null if nothing is
+     * selected
+     */
+    public static JsonNode loadJsonPreferences(final String loadDir, String filePrefix) {
+        final Preferences prefs = NbPreferences.forModule(ApplicationPreferenceKeys.class);
+        final String userDir = ApplicationPreferenceKeys.getUserDir(prefs);
+        final File prefDir = new File(userDir, loadDir);
+        final String[] names;
+        final ObjectMapper mapper = new ObjectMapper();
+
+        // Store the load directory/prefix so that they can be used by deleteJsonPreference
+        currentDir = loadDir;
+        currentPrefix = filePrefix;
+
+        // Check the supplied directory for any files, if filePrefix was supplied, only files
+        // containing the prefix are returned. Return a list of filenames for the user to select from.
+        if (prefDir.isDirectory()) {
+            names = prefDir.list((File dir, String name) -> {
+                if (filePrefix.isEmpty()) {
+                    return name.toLowerCase().endsWith(FILE_EXT);
+                }
+                return (name.toLowerCase().startsWith(filePrefix) && name.toLowerCase().endsWith(FILE_EXT));
+            });
+        } else {
+            // Nothing to select from - return an empty list
+            names = new String[0];
+        }
+
+        // chop off ".json" from the filenames
+        for (int i = 0; i < names.length; i++) {
+            final String nextName = FilenameEncoder.decode(names[i].substring(0, names[i].length() - 5));
+            if (nextName != null && !nextName.isEmpty()) {
+                names[i] = nextName;
+                // Hide any file prefix which the user didn't see when saving
+                if (!filePrefix.isEmpty()) {
+                    names[i] = names[i].substring(filePrefix.length());
+                }
+            } 
+        }
+
+        // Allow user to select a filename from the crafted list using the dialog.
+        String fileName = JsonIODialog.getSelection(names);
+        if (fileName != null) {
+            try {
+                // Reconsitute filename to include any file prefix
+                if (!filePrefix.isEmpty()) {
+                    fileName = filePrefix.concat(fileName);
+                }
+                return mapper.readTree(new File(prefDir, FilenameEncoder.encode(fileName) + FILE_EXT));
+            } catch (IOException ex) {
+                Exceptions.printStackTrace(ex);
```

Comment:
is it possible is this can be made to use a logger instead?

Reply:
done.